### PR TITLE
Update itch from 25.5.1 to 25.6.1-canary

### PR DESCRIPTION
**Pull request by erenakgunduz**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by erenakgunduz** [comment]
This is a bit strange, they [claim to have] published a new version according to their Releases page but the download link on their website doesn't reflect this, nor does the update check within the application itself

https://github.com/itchio/itch/releases/tag/v25.6.1-canary

**Comment by miccal** [comment]
> This is a bit strange, they [claim to have] [published a new version](https://github.com/itchio/itch/releases/tag/v25.6.1-canary) according to their Releases page but the download link on their website doesn't reflect this, and neither does the update check within the application itself

The download link `https://broth.itch.ovh/install-itch/darwin-amd64/LATEST/archive/default` gives version `25.0.0`, so we need to come up with a `livecheck` that reflect this.

**Review by samford — COMMENTED** [review]
25.0.0 is the version of `Install itch.app` but the Itch version that's downloaded and installed is 25.5.1, which is the newest version listed on https://broth.itch.ovh/itch/darwin-amd64. That is to say, the existing `livecheck` block gives the correct latest version (unlike `ExtractPlist`).

File: Casks/itch.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
 cask "itch" do
-  version "25.5.1"
+  version "25.0.0"
```
Comment (samford) [inline Casks/itch.rb:2]:
```suggestion
  version "25.5.1"
```

File: Casks/itch.rb [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-  name "itch.io"
+  name "itch"
   desc "Game client for itch.io"
   homepage "https://itch.io/app"
 
   livecheck do
-    url "https://broth.itch.ovh/itch/darwin-amd64"
-    regex(/(\d+(?:\.\d+)+)/i)
+    url :url
+    strategy :extract_plist
```
Comment (samford) [inline Casks/itch.rb:13]:
```suggestion
    url "https://broth.itch.ovh/itch/darwin-amd64"
    regex(%r{href=.*?/v?(\d+(?:\.\d+)+)/?["' >]}i)
```

This reinstates the existing `livecheck` block while tightening up the regex a bit.

**Review by miccal — APPROVED** [review]

**Pull request merged**